### PR TITLE
Make check-zone also check whether there are duplicate key value pair metadatas for the zone

### PR DESCRIPTION
**Pull request by RobinGeuze**
### Short description
This PR is the initial work for fixing #2214, initially returning errors in check-zone for the offending values.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/pdnsutil.cc [hunk 1]
```diff
@@ -598,6 +598,27 @@ int checkZone(DNSSECKeeper &dk, UeberBackend &B, const DNSName& zone, const vect
     }
   }
 
+  std::map<std::string, std::vector<std::string>> metadatas;
+  if (!B.getAllDomainMetadata(zone, metadatas)) {
```
Comment (Habbie) [inline pdns/pdnsutil.cc]: For terrible reasons, this skips `PRESIGNED`, `NSEC3PARAM`, and `NSEC3NARROW`for gsqlbackend (but not for other backends) - but I don't think we can blame this PR for that!

File: pdns/pdnsutil.cc [hunk 2]
```diff
@@ -598,6 +598,27 @@ int checkZone(DNSSECKeeper &dk, UeberBackend &B, const DNSName& zone, const vect
...
+
+  for (const auto &metaData : metadatas) {
+    set<string> seen;
+    set<string> messaged;
+
+    for (const auto &value : metaData.second) {
+      if (seen.count(value) <= 0) {
+        seen.insert(value);
+      } else if (messaged.count(value) <= 0) {
+        cout << "[Error] Found duplicate metadata key value pair for zone " << zone << " with key " << metaData.first << " and value " << value << endl;
```
Comment (Habbie) [inline pdns/pdnsutil.cc]: `[Error] Found duplicate metadata key value pair for zone 1.example.com with key PUBLISH-CDS and value `


```suggestion
        cout << "[Error] Found duplicate metadata key value pair for zone " << zone << " with key '" << metaData.first << "' and value '" << value << "'" << endl;
```

File: pdns/pdnsutil.cc [hunk 3]
```diff
@@ -598,6 +598,27 @@ int checkZone(DNSSECKeeper &dk, UeberBackend &B, const DNSName& zone, const vect
...
+    cout << "[Error] Unable to retrieve metadata for zone " << zone << endl;
+    numerrors++;
+  }
+
+  for (const auto &metaData : metadatas) {
+    set<string> seen;
+    set<string> messaged;
+
+    for (const auto &value : metaData.second) {
+      if (seen.count(value) <= 0) {
```
Comment (Habbie) [inline pdns/pdnsutil.cc]: while not wrong, the `<` looks weird to me :)

**Comment by Habbie** [comment]
ping!

File: pdns/pdnsutil.cc (hunk 1 above)
Comment (RobinGeuze) [reply pdns/pdnsutil.cc:602]: How best to address the issue?

File: pdns/pdnsutil.cc (hunk 3 above)
Comment (RobinGeuze) [reply pdns/pdnsutil.cc:612]: Right, looking at the docs for count it can only be 0 or 1, so I will change it.

File: pdns/pdnsutil.cc (hunk 1 above)
Comment (RobinGeuze) [reply pdns/pdnsutil.cc:602]: So based on the code it seems that the only case where they are not included is if dnssec is disabled on the backend. Do we really care about the rare case where dnssec is disabled and there are DNSSEC params?

**Comment by RobinGeuze** [comment]
@Habbie could you take another look? There is one review comment open, however its covers a corner case we can probably ignore in my opinion, do you agree?

File: pdns/pdnsutil.cc (hunk 1 above)
Comment (Habbie) [reply pdns/pdnsutil.cc:602]: I've decided we don't :)

**Comment by Habbie** [comment]
I agree. Can you rebase?

**Comment by RobinGeuze** [comment]
Done.

**Review by Habbie — APPROVED** [review]

**Pull request merged**